src/Symfony/Bundle/FrameworkBundle/Console/Application.php

```diff
@@ -68,24 +68,29 @@ public function doRun(InputInterface $input, OutputInterface $output): int
     {
         $this->registerCommands();
 
+        $statusCode = Command::SUCCESS;
         if ($this->registrationErrors) {
+            $statusCode = Command::FAILURE;
+
             $this->renderRegistrationErrors($input, $output);
         }
 
         $this->setDispatcher($this->kernel->getContainer()->get('event_dispatcher'));
 
-        return parent::doRun($input, $output);
+        return max($statusCode, parent::doRun($input, $output));
```

Comment:
I think we should not change the status code of all existing commands.
Instead, I propose that we change the status code of the "list" command only.

Reply:
Hm, but why? Is there any reason any command should succeed when there are registration errors?